ladybug/analysisperiod.py

```diff
@@ -201,47 +221,110 @@ def st_hour(self):
 
+    @property
+    def timestep(self):
+        """Timestep."""
+        return self._timestep
+
     @property
     def is_leap_year(self):
         """A boolean to indicate if analysis period is for a leap year."""
         return self._is_leap_year
 
+    @property
+    def st_time(self):
+        """Start datetime."""
+        return self._st_time
+
+    @property
+    def end_time(self):
+        """End datetime."""
+        return self._end_time
+
     @property
     def datetimes(self):
         """A sorted list of datetimes in this analysis period."""
-        # sort dictionary based on key values (minute of the year)
+        if self._timestamps_data is None:
+            self._calculate_timestamps()
         return tuple(DateTime.from_moy(moy, self.is_leap_year)
                      for moy in self._timestamps_data)
 
+    @property
+    def moys(self):
+        """A sorted list of minutes of year in this analysis period as integers."""
+        if self._timestamps_data is None:
+            self._calculate_timestamps()
+        return self._timestamps_data
+
     @property
     def hoys(self):
         """A sorted list of hours of year in this analysis period."""
+        if self._timestamps_data is None:
+            self._calculate_timestamps()
         return tuple(moy / 60.0 for moy in self._timestamps_data)
 
     @property
     def hoys_int(self):
-        """A sorted list of hours of year values in this analysis period as integers."""
+        """A sorted list of hours of year in this analysis period as integers."""
+        if self._timestamps_data is None:
+            self._calculate_timestamps()
         return tuple(int(moy / 60.0) for moy in self._timestamps_data)
 
+    @property
+    def doys_int(self):
+        """A sorted list of days of the year in this analysis period as integers."""
+        if not self._is_reversed:
+            return self._calc_daystamps(self.st_time, self.end_time)
+        else:
+            doys_st = self._calc_daystamps(self.st_time, DateTime.from_hoy(8759))
+            doys_end = self._calc_daystamps(DateTime.from_hoy(0), self.end_time)
+            return doys_st + doys_end
+
+    @property
+    def months_int(self):
+        """A sorted list of months of the year in this analysis period as integers."""
+        if not self._is_reversed:
+            return list(xrange(self.st_time.month, self.end_time.month + 1))
+        else:
+            months_st = list(xrange(self.st_time.month, 13))
+            months_end = list(xrange(1, self.end_time.month + 1))
+            return months_st + months_end
+
+    @property
+    def months_per_hour(self):
+        """A list of tuples representing months per hour in this analysis period."""
+        month_hour_strings = []
+        hour_range = xrange(self.st_hour, self.end_hour + 1)
+        for month in self.months_int:
+            month_hour_strings.extend([(month, hr) for hr in hour_range])
+        return month_hour_strings
```

Comment:
Very good point.  The name was leftover from a previous idea that I was playing with before I realized that it would be much cleaner to do this with tuples.  I have taken out the "string" from the name.  As you probably see later in the code, this method is used as the fundamental way of tracking the datetimes for MonthlyPerHour collections.